rtloader/common/stringutils.c

```diff
@@ -44,10 +43,12 @@ char *as_string(PyObject *object)
         PyErr_Clear();
         return NULL;
     }
-    retval = strdupe(tmp);
-#else
-    PyObject *temp_bytes = NULL;
 
+    temp_bytes = object;
+    retval = tmp;
+    Py_INCREF(temp_bytes);
+    *stringObject = temp_bytes;
```

Comment:
I copied these parts from existing places nearly AS IS. I am not that expert on CPyton nuances.

Reply:
I did notice small difference between 2 and 3 and tried to replicate it instead of rewriting it without understanding full implications in CPython internals.